Ginger/GingerCoreNET/ValueExpressionLib/ValueExpression.cs

```diff
@@ -1502,19 +1479,48 @@ private void ReplaceVarWithValue(string p, string[] a)
 
             if (vb != null)
             {
-                VarValue = vb.Value;
+                if (vb is VariablePasswordString)
+                {
+                    string strValuetoPass;
```

Comment:
need to change code to handle 2 main cases:
1- having only "Name" key: then only replace value as now (including the encryption specific)
2- having more than Name key: remove the name from DIC and pass it to variable GetValueWithParam to handle